File: VB/CustomizedTrackBar/CustomRangeTrackBar/CustomRangeTrackBar.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace CustomizedRangeTrackBar
	Friend Class CustomRangeTrackBar
		Inherits RangeTrackBarControl
		Shared Sub New()
			RepositoryItemCustomRangeTrackBar.RegisterCustomRangeTrackBar()
		End Sub
		Public Sub New()
			MyBase.New()
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemCustomRangeTrackBar.CustomRangeTrackBarName
			End Get
		End Property

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemCustomRangeTrackBar
			Get
				Return TryCast(MyBase.Properties, RepositoryItemCustomRangeTrackBar)
			End Get
		End Property
	End Class
End Namespace

File: CS/CustomizedTrackBar/CustomRangeTrackBar/CustomRangeTrackBar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using DevExpress.XtraEditors;

namespace CustomizedRangeTrackBar
{
    class CustomRangeTrackBar : RangeTrackBarControl
    {
        static CustomRangeTrackBar() { RepositoryItemCustomRangeTrackBar.RegisterCustomRangeTrackBar(); }
        public CustomRangeTrackBar() : base() { }

        public override string EditorTypeName { get { return RepositoryItemCustomRangeTrackBar.CustomRangeTrackBarName; } }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomRangeTrackBar Properties
        { get { return base.Properties as RepositoryItemCustomRangeTrackBar; } }
    }
}
